src/common/atlas/apiClient.ts

```diff
@@ -252,6 +252,30 @@ export class ApiClient {
         }
     }
 
+    async getIpInfo() {
+        await this.validateToken();
+
+        const endpoint = "api/private/ipinfo";
+        const url = new URL(endpoint, config.apiBaseUrl);
+        const response = await fetch(url, {
+            method: "GET",
+            headers: {
+                Accept: "application/json",
+                Authorization: `Bearer ${this.token?.access_token}`,
```

Comment:
[nit] Assuming this.token is never null if `this.validateToken` returned successfully.

```suggestion
                Authorization: `Bearer ${this.token!.access_token}`,
```